### DISCUSSION TITLE
Suggestion: Have you had the opportunity to try the rare Matrox G550 PCIe series x1 GPU cards ?

### DISCUSSION THREAD

**Discussion by BegoneRaven** [geerlingguy/raspberry-pi-pcie-devices#187] [Ideas]
The [Matrox G550 PCIe](https://web.archive.org/web/20200805170655/https://www.matrox.com/graphics/en/products/graphics_cards/g_series/g550pcie/) & the [Matrox G550 LP PCIe](https://web.archive.org/web/20200805161704/https://www.matrox.com/graphics/en/products/graphics_cards/g_series/g550lppcie/) are probably some rarely used cards. 
In their website they claim:
> The Matrox G550 PCIe is a dual monitor PCI express graphics card with x1 performance and has Windows and **open source display drivers for Linux along with other Unix-like operating systems**.

I am not sure if drivers are available on the current kernel or we need to install them manually, but this lonely & basic PCIe x1 primarily 2d-card is out there..Because it's focus is 2d, probably it needs less PCI resources like bars to be initialized ???
![](https://web.archive.org/web/20190909204558im_/https://matrox.com/graphics/media/image/products/graphics_cards/g_series/g550pcie.jpg)

**Comment by 6by9** [comment]
There is the kernel config option FB_MATROX (drivers/video/fbdev/matrox) which lists
```
          Say Y here if you have a Matrox Millennium, Matrox Millennium II,
          Matrox Mystique, Matrox Mystique 220, Matrox Productiva G100, Matrox
          Mystique G200, Matrox Millennium G200, Matrox Marvel G200 video,
          Matrox G400, G450 or G550 card in your box.
```
However that may or may not support the PCIe version, and it was marked as orphaned in MAINTAINERS (ie no one is actively maintaining or testing it) back in 2010.
```
MATROX FRAMEBUFFER DRIVER
L:	linux-fbdev@vger.kernel.org
S:	Orphan
F:	drivers/video/fbdev/matrox/matroxfb_*
F:	include/uapi/linux/matroxfb.h
```
It's also against the older fbdev API instead of DRM/KMS, which limits how useful it would be.

With the low max resolution
```
Maximum resolutions (per display)
    Digital: 1280 x 1024
    Analog, main display: 2048 x 1536
    Analog, secondary display: 1600 x 1200
```
it also isn't so useful as that is lower than the internal graphics is capable of.

https://en.wikipedia.org/wiki/Matrox_G400
> On 2005-7-13, Matrox Graphics Inc. announced the availability of Millennium G550 PCIe, the world's first PCI Express x1 graphics card.[6] The card uses Texas Instruments XIO2000 bridge controller to achieve PCI Express support.

It depends on how the bridge chip was has to map the cards resources. All PCI cards I've tried on a PCIe->PCI bridge generally fail as they make use of I/O bars to map the x86 style I/O register space, but ARM doesn't support this (everything is memory mapped).

There is a DRM driver for G200/G400 cards (drivers/gpu/drm/mgag200), but again it was marked as orphaned and obsolete in 2016, with nothing significant having been done to it for a fair while before that.

So whilst a nice idea, I don't think it's worth investing the time in trying to test these cards.

**Reply by BegoneRaven** [reply]
@6by9 , Thanks for the investigation on the Matrox cards ! :)
